Mathlib/Data/Finsupp/Defs.lean

```diff
@@ -910,6 +910,20 @@ theorem support_add_eq [DecidableEq α] {g₁ g₂ : α →₀ M} (h : Disjoint
 theorem single_add (a : α) (b₁ b₂ : M) : single a (b₁ + b₂) = single a b₁ + single a b₂ :=
   (zipWith_single_single _ _ _ _ _).symm
 
+theorem support_single_add [DecidableEq α] {a : α} {b : M} {f : α →₀ M}
+    (ha : a ∉ f.support) (hb : b ≠ 0) : support (single a b + f) = insert a f.support := by
+  have H := support_single_ne_zero a hb
+  apply (support_add_eq _).trans
+  · rw [H, insert_eq]
+  · rwa [H, disjoint_singleton_left]
+
+theorem support_add_single [DecidableEq α] {a : α} {b : M} {f : α →₀ M}
+    (ha : a ∉ f.support) (hb : b ≠ 0) : support (f + single a b) = insert a f.support := by
+  have H := support_single_ne_zero a hb
```

Comment:
This one should have a shorter proof by dualising, right? (I'm guessing we don't have add_comm here)

Reply:
Yes, this works even without commutative addition. Is there any type alias for reversing the order of addition?

Reply:
`AddOpposite`